File: BOJ/python/17204.py
import sys

input = sys.stdin.readline

N, K = map(int, input().split(' '))

global game_of_death
global check
global depth
game_of_death = [int(input()) for _ in range(N)]
check = [0 for _ in range(len(game_of_death))] # 지목 당한 적이 있는지 체크하기 위한 list
depth = [-1 for _ in range(len(game_of_death))] # 몇 번째로 지목 되었는지 확인하기 위한 list


def playGame(cur, d):
    global game_of_death
    global check
    global depth

    next = game_of_death[cur] 

    # 다음에 지목할 사람이 이미 지목된 적이 있으면 
    # 계속 cycle이 돌 것이기 때문에 더 이상 진행하는 것은 의미가 없다.
    if check[cur] == 1:
        return

    # 지목 된 적이 없으면 게임을 계속 진행한다.
    check[cur] = 1
    depth[cur] = d
    playGame(next, d + 1)


playGame(0, 0)
print(depth[K])

File: BOJ/17204.cpp
#include<bits/stdc++.h>
using namespace std;

int N, K;
int death_game[150];
int check[150]; // 지목 당한 적이 있는지 확인하기 위한 배열
int depth[150]; // 몇 번째로 처음 지목 당했는지 저장하기 위한 배열

void Reset() {
	for (int i = 0; i < 150; i++) {
		depth[i] = -1;
	}
}

void solve(int cur, int d) {
	int next = death_game[cur];
	if (check[cur] == 1) return; // 이미 이전에 지목당한 적이 있으면 함수를 종료한다.

	// 지목당한 적이 없으면 게임을 계속 진행한다.
	check[cur] = 1;
	depth[cur] = d;
	solve(next, d + 1);
}

int main() {
	Reset(); // depth배열 초기화

	cin >> N >> K;
	for (int i = 0; i < N; i++) {
		cin >> death_game[i];
	}

	solve(0, 0);

	// 지목당한 적이 없으면 배열의 초기화 값인 -1일 것이므로 지목당하지 않는 경우도 만족한다.
	cout << depth[K] << '\n';

	return 0;
}